Project-18-Food-order-app/src/components/Checkout/index.jsx

```diff
@@ -0,0 +1,162 @@
+import { useContext, useEffect } from "react";
+import PropTypes from "prop-types";
+
+import { CartContext } from "../../services/stores/CartContext";
+import { currencyFormatter } from "../../services/utils/currencyFormat";
+import { UserProgressContext } from "../../services/stores/UserProgress";
+
+import style from "./checkout.module.scss";
+import Button from "../UI/Button";
+import Input from "../UI/Input";
+import useInput from "../../hooks/useInput";
+import {
+  isEmail,
+  isNotEmpty,
+  isAllLetter,
+} from "../../services/utils/inputValidate";
+import { URL } from "../../services/constants/constants";
+import useHttp from "../../hooks/useHttp";
+import Error from "../UI/Error";
+
+const requestConfig = {
+  method: "POST",
+  headers: {
+    "Content-Type": "application/json",
+  },
+};
+
+function Checkout({ onClose }) {
+  const { hideCheckout, showSuccessMessageDialog } =
+    useContext(UserProgressContext);
+
+  const { cartData, clearCartData } = useContext(CartContext);
+
+  const {
+    value: emailValue,
+    handleInputChange: handleEmailChange,
+    handleInputBlur: handleEmailBlur,
+    hasError: hasEmailError,
+  } = useInput("", (value) => isEmail(value) && isNotEmpty(value));
+
+  const {
+    value: nameValue,
+    handleInputChange: handleNameChange,
+    handleInputBlur: handleNameBlur,
+    hasError: hasNameError,
+  } = useInput("", (value) => isAllLetter(value));
+
+  const cartTotal = cartData.reduce(
+    (totalPrice, item) => totalPrice + +item.price * item.quantity,
+    0
+  );
+
+  const handleCloseButtonClick = () => {
+    hideCheckout();
+    onClose();
+  };
+
+  const {
+    data,
+    isLoading: isSending,
+    error,
+    sendRequest,
+  } = useHttp(`${URL}orders`, requestConfig);
+
+  const handleSubmit = (e) => {
```

Comment:
@sknepali You have handled the error case in your form submission, which is good. However, you might want to consider resetting the error state once the form is successfully submitted.